packages/components/src/components/primitives/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC } from 'react';
+import {
+  PlacementWithLogical,
+  Tooltip as ChakraTooltip,
+  chakra,
+} from '@chakra-ui/react';
+import { TooltipProps } from './types';
+
+const chakraPlacementPrefixToComposerPositioning = {
+  left: 'start',
+  right: 'end',
+};
+
+export const calculateTooltipPlacement = (
+  side: string,
+  positioning: string
+) => {
+  if (positioning === 'center') {
+    return side;
+  }
+
+  return `${side}-${chakraPlacementPrefixToComposerPositioning[positioning]}`;
+};
+
+export const Tooltip: FC<TooltipProps> = ({
+  side,
+  positioning,
+  children,
+  ...props
+}) => {
+  const placement = calculateTooltipPlacement(
+    side,
+    positioning
+  ) as PlacementWithLogical;
+
+  return (
+    <ChakraTooltip
+      data-testid="cmpsr.tooltip"
+      hasArrow
+      placement={placement}
+      {...props}
+    >
+      <chakra.span h="full" display="inline-block">
```

Comment:
why do we need this wrapper?

Reply:
because without adding it, Chakra won't be able to inject the ref inside the component to set the anchor for the relative positioning of the tooltip